bugfix, if framesize>168 then data is cutoff when using odd length fo…

A1337CBS opened the pull request:
Hello,

Please check this, I was trying to send data where payload>168 bits and noticed that I wasn't receiving anything but it worked fine when payload<168 bits. So if you check the code you will see you have no check to make LEN_FRAME even so that it can later be converted to bytes for byte_Frame which is defined just two lines below.

Second commit is not really needed.

A1337CBS committed:
bugfix, if framesize>168 then data is cutoff when using odd length for LEN_FRAME

zwh2 commented:
I see the problem you are having but wouldn't a simpler solution be to just add 7 before the divide on line 428 so it would read `unsigned char byte_frame[(LEN_FRAME+7)/8];`

Could you provide some clarification on the purpose of 24aab3f

A1337CBS commented:
Yea, that could work too, for me using while was the simpler approach though.

[24aab3f](https://github.com/trendmicro/ais/commit/24aab3f9a33b98bbffc89177a999cae75f8e1fc9) -> This commit is not needed, I will remove it

A1337CBS commented:
@zwh2 Commit removed, can you merge the pull request now? 

I have also a couple of other changes I would like to discuss, for example I created a new block to add support for taking input from socket to the AIS_frame_builder

zwh2 commented:
Unfortunately I do not have merge permissions on this repo. I would be interested in what you have done to add taking input from a socket.

A1337CBS commented:
@zwh2 Oh my bad, I thought you did. Well I changed the block type to tagged stream block, added an option to take input to the block itself or use the old way and oh I re-made the whole block using the latest 3.7 version of gnuradio so everything is updated. If you will like to use it let me know and I will upload it to my repo.

@embyte0  Hi, will you be willing to add this and the above?

zwh2 commented:
I think that would be very interesting to take a look at. Also making it available to others may help them as sometimes it takes embyte0 a while to be able to update this repo

embyte0 commented:
Hi folks, I am currently on trip but I will look into this next week. Appreciated your work.
